### DISCUSSION TITLE
Next gen Auro design tokens support architecture

### DISCUSSION THREAD

**Discussion by blackfalcon** [AlaskaAirlines/WC-Generator#539] [General Auro discussions]
## The problem

The Auro design system continues to evolve with the business needs and in turn components of this system need to evolve as well. 

The current UI foundational tokens are no longer supporting the business needs and a new system of foundational tokens are being designed. Additionally the previous concept of individual teams managing their own set of tokens for theme support is no longer the direction of the org. It is requested that Auro design tokens not only support the default UI, but all the other themes as well. 

This requires a re-thinking architecturally of how to support these growing needs of the teams and the business. 

## The recommendation 

TBD...

**Comment by fajar-apri-alaska** [comment]
Hi @blackfalcon,
I'm curious, what is the meaning of 

> "It is requested that Auro design tokens not only support the default UI, but all the other themes as well."

if we compare it with Auro current implementation ?

Maybe a token generator with realtime UI would be a way to help ?

**Reply by blackfalcon** [reply]
That line suggests that the Auro Design Token repo won't just have the JSON tokens for the default Auro UI but also include the main themes of the business.

Right now, the tricky part isn't creating a token generator tool, but rather defining the themes themselves. Once teams get the hang of it, we can think about automating tools to handle this stuff. But for now, things are still pretty up in the air.

**Comment by blackfalcon** [comment]
This commit adds functionality to the Auro Design Token build process to generate two new JSON data types of processed token exports. 

https://github.com/AlaskaAirlines/AuroDesignTokens/pull/148/commits/b9918b4c7a280a0cb69a4b31cc770a13e7275048

**Reply by blackfalcon** [reply]
In addition, using this URL will allow any consumer pull down the JSON file without having to install and extract from a npm resource. 

https://cdn.jsdelivr.net/npm//@aurodesignsystem/design-tokens@latest/dist/tokens/JSONVariablesFlat.json

NOTE: The update to produce this asset is NOT YET released. This URL will return an error until the PR is released. 

Also, the `@latest` should not be used as a production endpoint. Please be sure to update with the version you intend to download.
